app/controllers/state_file/questions/data_review_controller.rb

```diff
@@ -8,7 +8,7 @@ def edit
            current_intake&.disqualifying_df_data_reason.present?
           redirect_to next_path and return
         end
-        if current_intake&.df_data_import_failed_at.present?
```

Comment:
I think that case would be caught by `current_intake&.has_disqualifying_eligibility_answer? || current_intake&.disqualifying_df_data_reason.present?` in the conditional above. `df_data_import_failed_at` if there is a system error thrown during the import. I was considering making this field a list so we can track how many attempts are made per intake and when they happened rather than updating or clearing it.

Reply:
Oh yes I think you're right in that case current_intake&.disqualifying_df_data_reason would return :has_out_of_state_w2

I guess it would just be in the case that anything in this `begin` block fails which could be more than just the `raw_direct_file_data` not updating but probably unlikely right? We could check that it has updated the other things correctly like federal_submission_id, hashed_ssns, dependents etc but maybe that's overkill. Idk your call
```
begin
        direct_file_json = IrsApiService.import_federal_data(authorization_code, intake.state_code)

        intake.update(
          raw_direct_file_data: direct_file_json['xml'],
          federal_submission_id: direct_file_json['submissionId'],
          federal_return_status: direct_file_json['status']
        )
        intake.update(
          hashed_ssn: SsnHashingService.hash(intake.direct_file_data.primary_ssn)
        )
        intake.synchronize_df_dependents_to_database
      rescue => err
        Rails.logger.error(err)
        intake.update(df_data_import_failed_at: DateTime.now)
      end
      ```

Reply:
good point. now I'm thinking about adding more log messages to indicate specifically if any of those end up missing from the response, and perhaps going back to using `df_data_import_failed_at` as the indicator but just clearing it if that block succeeds.

Reply:
@mpidcock yeah that seems fine for now, we can re-evaluate later